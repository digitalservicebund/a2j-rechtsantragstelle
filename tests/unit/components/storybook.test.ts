describe("snapshot test", () => {
  test.todo(
    "decide about snapshot tests, see https://storybook.js.org/docs/writing-tests/storyshots-migration-guide",
  );
});
