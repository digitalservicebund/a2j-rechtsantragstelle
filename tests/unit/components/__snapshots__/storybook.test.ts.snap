// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button/Button Ghost 1`] = `
<button
  className="ds-button ds-button-ghost"
>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
</button>
`;

exports[`Storyshots Button/Button Icon Left 1`] = `
<button
  className="ds-button ds-button-with-icon"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium ds-button-icon  css-i4bv87-MuiSvgIcon-root"
    data-testid="HomeOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="m12 5.69 5 4.5V18h-2v-6H9v6H7v-7.81l5-4.5M12 3 2 12h3v8h6v-6h2v6h6v-8h3L12 3z"
    />
  </svg>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
</button>
`;

exports[`Storyshots Button/Button Icon Right 1`] = `
<button
  className="ds-button ds-button-with-icon"
>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium ds-button-icon  css-i4bv87-MuiSvgIcon-root"
    data-testid="HomeOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="m12 5.69 5 4.5V18h-2v-6H9v6H7v-7.81l5-4.5M12 3 2 12h3v8h6v-6h2v6h6v-8h3L12 3z"
    />
  </svg>
</button>
`;

exports[`Storyshots Button/Button Large 1`] = `
<button
  className="ds-button ds-button-large"
>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
</button>
`;

exports[`Storyshots Button/Button Primary 1`] = `
<button
  className="ds-button"
>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
</button>
`;

exports[`Storyshots Button/Button Secondary 1`] = `
<button
  className="ds-button ds-button-secondary"
>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
</button>
`;

exports[`Storyshots Button/Button Small 1`] = `
<button
  className="ds-button ds-button-small"
>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
</button>
`;

exports[`Storyshots Button/Button Tertiary 1`] = `
<button
  className="ds-button ds-button-tertiary"
>
   
  <span
    className="ds-button-label"
  >
    Button
  </span>
   
</button>
`;

exports[`Storyshots Button/ButtonContainer Two Buttons 1`] = `
<div
  className="flex flex-wrap gap-24"
>
  <button
    className="ds-button"
  >
     
    <span
      className="ds-button-label"
    >
      Button 1
    </span>
     
  </button>
  <button
    className="ds-button ds-button-secondary"
  >
     
    <span
      className="ds-button-label"
    >
      Button 2
    </span>
     
  </button>
</div>
`;

exports[`Storyshots Component/Checkbox Default 1`] = `
<div>
  <input
    name="name"
    type="hidden"
    value="off"
  />
  <input
    className="ds-checkbox "
    defaultChecked={false}
    id="name"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    required={false}
    type="checkbox"
    value="value"
  />
  <label
    htmlFor="name"
  >
    <div
      className="rich-text ds-stack-8 "
      dangerouslySetInnerHTML={
        {
          "__html": "<p class="text-lg">label</p>",
        }
      }
    />
  </label>
</div>
`;

exports[`Storyshots Component/DateInput Default 1`] = `
[
  <h2>
    Unexpected Application Error!
  </h2>,
  <h3
    style={
      {
        "fontStyle": "italic",
      }
    }
  >
    Cannot read properties of null (reading 'isContentEditable')
  </h3>,
  <pre
    style={
      {
        "backgroundColor": "rgba(200,200,200, 0.5)",
        "padding": "0.5rem",
      }
    }
  >
    TypeError: Cannot read properties of null (reading 'isContentEditable')
    at new isContentEditable (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/imask/src/controls/input.js:52:11)
    at Object.IMask [as default] (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/imask/src/core/holder.js:11:10)
    at MaskedComponent.initMask (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-imask/src/mixin.ts:201:27)
    at MaskedComponent.initMask [as componentDidMount] (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-imask/src/mixin.ts:167:12)
    at commitLayoutEffectOnFiber (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:13151:28)
    at commitLayoutMountEffects_complete (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:14330:9)
    at commitLayoutEffects_begin (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:14316:7)
    at commitLayoutEffects (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:14301:3)
    at commitRootImpl (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:16062:5)
    at commitRoot (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:15933:5)
    at performSyncWorkOnRoot (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:15454:3)
    at flushSyncCallbacks (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:2597:22)
    at flushSyncCallbacksOnlyInLegacyMode (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:2576:5)
    at scheduleUpdateOnFiber (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:14910:7)
    at updateContainer (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:17770:5)
    at create (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/react-test-renderer/cjs/react-test-renderer.development.js:18510:3)
    at getRenderedTree (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/@storybook/addon-storyshots/dist/frameworks/react/renderTree.js:12:18)
    at /Users/frede/Documents/a2j-rechtsantragstelle/node_modules/@storybook/addon-storyshots/dist/test-bodies.js:9:24
    at Object.&lt;anonymous&gt; (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/@storybook/addon-storyshots/dist/api/snapshotsTestsTemplate.js:20:29)
    at Promise.then.completed (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/utils.js:298:28)
    at new Promise (&lt;anonymous&gt;)
    at callAsyncCircusFn (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/utils.js:231:10)
    at _callCircusTest (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/run.js:316:40)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at _runTest (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/run.js:252:3)
    at _runTestsForDescribeBlock (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/run.js:126:9)
    at _runTestsForDescribeBlock (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/run.js:121:9)
    at _runTestsForDescribeBlock (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/run.js:121:9)
    at run (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/run.js:71:3)
    at runAndTransformResultsToJestFormat (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)
    at jestAdapter (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
    at runTestInternal (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-runner/build/runTest.js:367:16)
    at runTest (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-runner/build/runTest.js:444:34)
    at Object.worker (/Users/frede/Documents/a2j-rechtsantragstelle/node_modules/jest-runner/build/testWorker.js:106:12)
  </pre>,
  <p>
    ðŸ’¿ Hey developer ðŸ‘‹
  </p>,
  <p>
    You can provide a way better UX than this when your app throws errors by providing your own 
    <code
      style={
        {
          "backgroundColor": "rgba(200,200,200, 0.5)",
          "padding": "2px 4px",
        }
      }
    >
      ErrorBoundary
    </code>
     or
     
    <code
      style={
        {
          "backgroundColor": "rgba(200,200,200, 0.5)",
          "padding": "2px 4px",
        }
      }
    >
      errorElement
    </code>
     prop on your route.
  </p>,
]
`;

exports[`Storyshots Container/Background Background Color 1`] = `
<div
  className="bg-blue-100 !pt-0 !pb-0"
>
  Lorem ipsum
</div>
`;

exports[`Storyshots Container/Background Default 1`] = `
<div
  className="!pt-0 !pb-0"
>
  Lorem ipsum
</div>
`;

exports[`Storyshots Container/Background Paddings 1`] = `
<div
  className="!pt-32 !pb-48"
>
  Lorem ipsum
</div>
`;

exports[`Storyshots Container/Container Background Color 1`] = `
<div
  className="container !pt-40 !pb-48 text-black bg-blue-100"
>
  Lorem ipsum
</div>
`;

exports[`Storyshots Container/Container Custom Paddings 1`] = `
<div
  className="container !pt-8 !pb-16"
>
  Lorem ipsum
</div>
`;

exports[`Storyshots Container/Container Default 1`] = `
<div
  className="container !pt-40 !pb-48"
>
  Lorem ipsum
</div>
`;

exports[`Storyshots Container/Container Overhanging Background 1`] = `
<div
  className="overflow-x-hidden rounded-lg"
  style={
    {
      "marginLeft": "clamp(1rem, 5vw, 3rem)",
      "marginRight": "clamp(1rem, 5vw, 3rem)",
    }
  }
>
  <div
    className="container !pt-40 !pb-48 text-black relative before:content-[''] before:absolute before:inset-y-0 before:-left-32 before:-right-32 before:rounded-lg before:bg-blue-100"
  >
    <div
      className="relative"
    >
      Lorem ipsum
    </div>
  </div>
</div>
`;

exports[`Storyshots Content/Box Default 1`] = `
<div
  className="bg-blue-100 !pt-32 !pb-32"
>
  <div
    className="container !pt-32 !pb-40 text-black bg-yellow-300"
  >
    <div
      className="ds-stack-16 scroll-my-40"
      id="default-box-id"
    >
      <div
        className="ds-stack-8"
      >
        <p
          className="ds-label-02-reg"
          dangerouslySetInnerHTML={
            {
              "__html": "Label text",
            }
          }
        />
        <h2
          className="ds-heading-03-bold"
          dangerouslySetInnerHTML={
            {
              "__html": "Heading text",
            }
          }
        />
        <div>
          <div
            className="rich-text ds-stack-8 "
            dangerouslySetInnerHTML={
              {
                "__html": "<p class="text-lg">Lorem <strong>ipsum</strong></p><ul>
<li>Lorem ipsum</li>
<li>Lorem ipsum</li>
</ul>
",
              }
            }
          />
        </div>
      </div>
      <div
        className="flex flex-wrap gap-24"
      >
        <button
          className="ds-button ds-button-tertiary"
        >
           
          <span
            className="ds-button-label"
          >
            Button 1
          </span>
           
        </button>
        <button
          className="ds-button"
        >
           
          <span
            className="ds-button-label"
          >
            Button 2
          </span>
           
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Content/UserFeedback Example 1`] = `
<div
  className="bg-white !pt-32 !pb-40"
>
  <div
    className="overflow-x-hidden rounded-lg"
    style={
      {
        "marginLeft": "clamp(1rem, 5vw, 3rem)",
        "marginRight": "clamp(1rem, 5vw, 3rem)",
      }
    }
  >
    <div
      className="container !pt-32 !pb-32 text-black relative before:content-[''] before:absolute before:inset-y-0 before:-left-32 before:-right-32 before:rounded-lg before:bg-blue-300"
    >
      <div
        className="relative"
      >
        <div
          className="ds-stack-16"
          data-testid="user-feedback-banner"
        >
          <h2
            className="ds-label-01-bold"
            dangerouslySetInnerHTML={
              {
                "__html": "Hat Ihnen der Vorab-Check geholfen?",
              }
            }
          />
          <form
            action="/action/send-rating?url=/&context=&js=false"
            method="post"
            onSubmit={[Function]}
          >
            <div
              className="flex flex-wrap gap-24"
            >
              <button
                className="ds-button ds-button-tertiary ds-button-with-icon"
                name="wasHelpful"
                type="submit"
                value="yes"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium ds-button-icon  css-i4bv87-MuiSvgIcon-root"
                  data-testid="ThumbUpOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 21h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.58 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2zM9 9l4.34-4.34L12 10h9v2l-3 7H9V9zM1 9h4v12H1z"
                  />
                </svg>
                 
                <span
                  className="ds-button-label"
                >
                  Ja
                </span>
                 
              </button>
              <button
                className="ds-button ds-button-tertiary ds-button-with-icon"
                name="wasHelpful"
                type="submit"
                value="no"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium ds-button-icon  css-i4bv87-MuiSvgIcon-root"
                  data-testid="ThumbDownOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15 3H6c-.83 0-1.54.5-1.84 1.22l-3.02 7.05c-.09.23-.14.47-.14.73v2c0 1.1.9 2 2 2h6.31l-.95 4.57-.03.32c0 .41.17.79.44 1.06L9.83 23l6.59-6.59c.36-.36.58-.86.58-1.41V5c0-1.1-.9-2-2-2zm0 12-4.34 4.34L12 14H3v-2l3-7h9v10zm4-12h4v12h-4z"
                  />
                </svg>
                 
                <span
                  className="ds-button-label"
                >
                  Nein
                </span>
                 
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Page/Footer Example 1`] = `
<footer
  className="text-base"
>
  <div
    className="container !pt-48 !pb-48"
  >
    <div
      className="flex flex-wrap items-start justify-between gap-y-32"
    >
      <div
        className="flex flex-col flex-col-reverse sm:flex-row gap-y-8 gap-x-16"
      >
        <div>
          <img
            alt=""
            src="https://placehold.co/240x132/png"
            width={120}
          />
        </div>
        <div
          className="ds-stack-8"
        >
          <div>
            <div
              className="rich-text ds-stack-8 "
              dangerouslySetInnerHTML={
                {
                  "__html": "<p class="leading-snug">Lorem ipsum 1 <a class="text-link increase-tap-area whitespace-nowrap" href=/ target="_blank" rel="noopener">Link</a></p>",
                }
              }
            />
          </div>
          <div>
            <div
              className="rich-text ds-stack-8 "
              dangerouslySetInnerHTML={
                {
                  "__html": "<p class="leading-snug">Lorem ipsum 2 <a class="text-link increase-tap-area whitespace-nowrap" href=/ target="_blank" rel="noopener">Link</a></p>",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="flex flex-wrap gap-x-32 gap-y-8"
      >
        <ul
          className="list-none m-0 p-0 ds-stack-8"
        >
          <li
            className="leading-snug"
          >
            <a
              className="text-link increase-tap-area"
              href="https://digitalservice.bund.de"
              rel="noopener"
              target="_blank"
            >
              DigitalService
            </a>
          </li>
          <li
            className="leading-snug"
          >
            <a
              className="text-link increase-tap-area"
              href="/url1"
            >
              Lorem ipsum 1
            </a>
          </li>
          <li
            className="leading-snug"
          >
            <a
              className="text-link increase-tap-area"
              href="/url2"
            >
              Lorem ipsum 2
            </a>
          </li>
        </ul>
        <ul
          className="list-none m-0 p-0 ds-stack-8"
        >
          <li
            className="leading-snug"
          >
            <a
              className="text-link increase-tap-area"
              href="/url3"
            >
              Lorem ipsum 3
            </a>
          </li>
          <li
            className="leading-snug"
          >
            <a
              className="text-link increase-tap-area"
              href="/url4"
            >
              Lorem ipsum 4
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</footer>
`;
